src/shogun/machine/BaggingMachine.cpp

```diff
@@ -44,17 +47,53 @@ CBaggingMachine::~CBaggingMachine()
 
 CBinaryLabels* CBaggingMachine::apply_binary(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> output = apply_outputs_without_combination(data);
+
+	CMeanRule* mean_rule = new CMeanRule();
```

Comment:
You should use it if you created an object that is not passed to any other method.
If you create say `CBinaryLabels` and then pass it to a method `set_labels` this will increase the ref-counter by one, and the object that you called `set_labels` on will later on decrease the counter, which means the object will be deleted.
If you never pass it around, or increased the counter yourself, you need to unref later. If you don't see leaks, then nothing to worry